scripts/ccpp_track_variables.py

```diff
@@ -0,0 +1,194 @@
+
+        new_metadata_headers = parse_metadata_file(scheme_filename,
+                                                   known_ddts=registered_fortran_ddt_names(), run_env=run_env)
+        for scheme_metadata in new_metadata_headers:
+            for section in scheme_metadata.sections():
+                found_var = []
+                intent = ''
+                for scheme_var in section.variable_list():
+                    exact_match = False
+                    if var == scheme_var.get_prop_value('standard_name'):
+                        run_env.logger.debug(f"Found variable {var} in scheme {section.title}")
+                        found_var=var
+                        exact_match = True
+                        intent = scheme_var.get_prop_value('intent')
+                        break
+                    scheme_var_standard_name = scheme_var.get_prop_value('standard_name')
+                    if scheme_var_standard_name.find(var) != -1:
+                        run_env.logger.debug(f"{var} matches {scheme_var_standard_name}")
+                        found_var.append(scheme_var_standard_name)
+                if not found_var:
+                    run_env.logger.debug(f"Did not find variable {var} in scheme {section.title}")
+                elif exact_match:
+                    run_env.logger.debug(f"Exact match found for variable {var} in scheme {section.title},"
+                                  f" intent {intent}")
+                    var_graph.append((section.title,intent))
+                else:
+                    run_env.logger.debug(f"Found inexact matches for variable(s) {var} "
+                                  f"in scheme {section.title}:\n{found_var}")
+                    partial_matches[section.title] = found_var
+    if var_graph:
+        success = True
+        run_env.logger.debug(f"Successfully generated variable graph for sdf {suite.sdf_name}\n")
+    else:
+        success = False
+        run_env.logger.error(f"Variable {var} not found in any suites for sdf {suite.sdf_name}\n")
+        if partial_matches:
+            print("Did find partial matches that may be of interest:\n")
+            for key in partial_matches:
+                print(f"In {key} found variable(s) {partial_matches[key]}")
+
+    return (success,var_graph)
+
+def main():
+    """Main routine that traverses a CCPP suite and outputs the list of schemes that use given variable"""
+
+    args = parse_arguments()
+
+    logger = setup_logging(args.debug)
+
+    #Use new capgen class CCPPFrameworkEnv 
+    run_env = CCPPFrameworkEnv(logger, host_files="", scheme_files="", suites="")
+
+    suite = parse_suite(args.sdf,run_env)
+
+    (success, config) = import_config(args.config, None)
+    if not success:
+        raise Exception('Call to import_config failed.')
+
+    # Variables defined by the host model; this call is necessary because it converts some old
+    # metadata formats so they can be used later in the script
```

Comment:
I looked at `gather_variable_definitions` and it seems to be a function that doesn't act on any input data structures in place, or have any side effects. Perhaps you are calling this only for the data checking, but I really don't think that this call is doing anything to metadata that is used downstream, so the comment seems misleading. To have it convert metadata, I think you'd need to save its outputs in a data structure. Perhaps update one of the `config` entries? I'm not sure how that would work since I haven't dug into the details of the objects being acted on here.